pyfr/plugins/fluidforce.py

```diff
@@ -172,6 +213,13 @@ def __call__(self, intg):
 
                 # Do the quadrature
                 f[ndims:] += np.einsum('i...,klij,jil', qwts, vis, norms)
+                if self.morigin:
+                    # Viscous force at each flux point
+                    viscf = np.einsum('ijkl, lkj->lki', vis, norms)
+
+                    # Viscous force moments
+                    m[self._mcomponents:] += np.einsum(
+                        'i,ji', qwts, np.cross(rfpts, viscf)).sum()
```

Comment:
Should be able to merge the sum into the einsum expression.  Can also move the cross into the einsum which enables viscf to be merged, too.

Reply:
I'll wait for a bit more validation before merging the cross product in the `einsum`. Do you know of a quick way to use `einsum` for cross products without calling `tensordot`?

Reply:
It's been a while, but I believe I've used it in the past for batch cross product calculations.